lib/puppet/network/http/compression.rb

```diff
@@ -20,7 +20,8 @@ module Active
     def uncompress_body(response)
       case response['content-encoding']
       when 'gzip'
-        return Zlib::GzipReader.new(StringIO.new(response.body)).read
+        # ZLib::GzipReader has an associated encoding, by default Encoding.default_external
+        return Zlib::GzipReader.new(StringIO.new(response.body), :encoding => Encoding::BINARY).read
```

Comment:
I know that it's superfluous, but what do you think about adding in the explicit encoding to the `StringIO` initializer like:

```ruby
return Zlib::GzipReader.new(StringIO.new(response.body, 'rb'), :encoding => Encoding::BINARY).read
``` 

A couple reasons:

* If we do find an analysis tool to scan for missing usage of explicit encodings (RubyMine inspections might be our best bet - need to look into it more), then this would be one less thing to worry about.
* If for some reason `response.body` changes from a `BINARY` string, this could all of a sudden start failing hard - which would be a clear signal of a behavior change.